### ISSUE TITLE
Improve the experience for using SQLite in-memory databases

### ISSUE THREAD

**Issue by ajcvickers** [aspnet/EntityFrameworkCore#16103]
For example, some sugar to make it easier to keep a connection open across multiple context instances.

@divega @bricelam I believe you guys have some additional thoughts here--maybe link some of the information from Brice's blog?

**Comment by ArthurHNL** [comment]
Maybe also document which features need to be implemented "manually", for example I had to specify a standard default value for concurrency tokens, a converter for concurrency tokens, a comparer for concurrency tokens and a converter for datetimeoffssets

**Comment by musicm122** [comment]
Documentation on Registering with Dependency Injection and maintaining\ re-hydrating state across multiple tests would be useful as well.

**Comment by Timovzl** [comment]
I only just spotted this and haven't tried it, but perhaps it's made for this scenario:
[Sharable in-memory](https://docs.microsoft.com/en-us/dotnet/standard/data/sqlite/connection-strings#sharable-in-memory)

```
Data Source=Sharable;Mode=Memory;Cache=Shared
```

**Comment by Ninjanaut** [comment]
It's probably obvious, but there is no mention which Nuget package to install if someone want to use EF Sqlite in-memory feature.

**Comment by ajcvickers** [comment]
@Ninjanaut It doesn't require a different NuGet package; it's a feature built in to SQLite.

**Comment by sguryev** [comment]
I have created method in base test class:
`protected AppDbContext GetDbContext() => new AppDbContext(_dbContextOptions);`
So any time I need new context I call it.

I have came up to the statement that I need fresh context most of the time to avoid caching in `Local` and pass missing `.Include()` calls. Are there some more complicated connection management tasks?

**Comment by Yaevh** [comment]
the documentation (https://docs.microsoft.com/en-us/ef/core/testing/sqlite) is IMHO misleading re using SQLite in-memory database in conjunction with `.AddDbContext()` method, see #25204

**Comment by user1568891** [comment]
> It's probably obvious, but there is no mention which Nuget package to install if someone want to use EF Sqlite in-memory feature.

I agree. For someone new to sqlite just doing a Nuget Package Manager search for sqlite returns a lot of results. It really wouldn't hurt to note that Microsoft.EntityFrameworkCore.Sqlite is required.
